### DISCUSSION TITLE
Changing property of object in array

### DISCUSSION THREAD

**Discussion by JokerMartini** [quasarframework/quasar#10227] [General - Components / Directives / etc]
I have an array of objects in my state file. Each object contains very basic params like the following
```
{
   name: "JohnDoe",
   liked: false,
   image: "https://.....image.jpg"
}
```

I loop through this array of objects and display them using q-cards. Within each q-card it displays the image and checkbox(liked icon). How can I have it so when the user toggles the property it updates the actual value of the object's property? I tried to do this with actions but it's causing a flicker presumably because it's reloading the thumbnail. Below is the important bits of code....

![image](https://user-images.githubusercontent.com/1416749/127386135-8edf0233-a9ee-4cac-ae5d-d5a2d55823a0.png)

gallery.vue
```
<template>
  <q-page>
    <AssetCard
      v-for="x in assetList"
      :key="x"
      :item="x"
      @dblclick="showAssetOverview = true"
    >
  </q-page>
</template>
```

assetCard.vue
```
const likeToggled = () => {
      $store.dispatch("app/likeToggled", 
      {
        uid: props.item.uid,
        liked: !props.item.liked
      });
    };
```

actions.js
```
export function likeToggled({ commit, state }, payload) {
  const assets = state.assetList.map((item) => {
    if (item.uid == payload.uid) {
      return { ...item, ...payload };
    } else return item;
  });
  commit("setAssets", assets);
}
```

**Comment by ontwikkelfabriek** [comment]
@JokerMartini 

Please to receive better support create a codpen.io that demo's your problem.
